components/playlist/browser/resources/components/playerSeeker.tsx

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+import styled from 'styled-components'
+import { color, font } from '@brave/leo/tokens/css'
+
+interface Props {
+  videoElementRef: React.RefObject<HTMLVideoElement>
+}
+
+const SeekerContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`
+
+const TimeContainer = styled.div`
+  display: grid;
+  grid-template-columns: auto auto;
+  justify-content: space-between;
+
+  color: ${color.text.secondary};
+  font: ${font.primary.xSmall.regular};
+`
+
+const StyledProgress = styled.progress`
+  --progress-background: #423eee; /* TODO(sko) Neither color.button.background or color.semantic.button.background exist for now */
+  --progress-bar-height: 12px;
+  --progress-stroke-thickness: 2px;
+
+  width: 100%;
+  height: var(--progress-bar-height);
+  appearance: none;
+  position: relative;
+
+  &::-webkit-progress-bar {
+    background: rgba(0, 0, 0, 0.1);
+    height: var(--progress-stroke-thickness);
+    border-radius: calc(var(--progress-stroke-thickness) * 0.5);
+    opacity: 1;
+    transform: translate(0, calc((var(--progress-bar-height) - 100%) / 2));
+  }
+  &::-webkit-progress-value {
+    border-radius: calc(var(--progress-stroke-thickness) * 0.5);
+    background: var(--progress-background);
+  }
+  // TODO(sko) Can we split this so that we can avoid creating new class too
+  // many times?
+  &::after {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: calc(
```

Comment:
This might be a good use case for `.attrs`

https://styled-components.com/docs/basics#attaching-additional-props

Basically, it sets the `style` property directly on the element, rather than creating a bunch of random classes (which styled-components really struggles with). In my (limited) testing, it's much, much faster.

Reply:
Ah yeah you're right, you won't be able to style the pseudo element with `.attrs`, sorry didn't notice the pseudo element. 

Probably fine as is, but you could also try setting some CSS variables on the element, and using them in the pseudo element, if you're interested in playing around?

Reply:
Yay, that seems to work! Do you mind taking another look?